server/service/osquery_utils/queries.go

```diff
@@ -905,6 +913,20 @@ func directIngestMunkiInfo(ctx context.Context, logger log.Logger, host *fleet.H
 	return ds.SetOrUpdateMunkiVersion(ctx, host.ID, rows[0]["version"])
 }
 
+func ingestKubequeryInfo(ctx context.Context, logger log.Logger, host *fleet.Host, rows []map[string]string) error {
+	if len(rows) != 1 {
+		logger.Log("component", "service", "method", "ingestKubequeryInfo", "warn",
+			fmt.Sprintf("kubernetes_info expected single result got %d", len(rows)))
+	}
+
+	host.Hostname = fmt.Sprintf("kubequery %s", rows[0]["cluster_name"])
+
+	// These values are not phorovided by kubequery
+	host.OsqueryVersion = "kubequery"
+	host.Platform = "kubequery"
```

Comment:
Q: Is this "made up platform" (made up as in not provided by osquery) a linux platform?
https://github.com/fleetdm/fleet/blob/acf60189ff150ed0bf8d80e932a06d6cb688c3e4/server/fleet/hosts.go#L245-L257

Reply:
Technically yes, but kubequery doesn't support most of the typical Linux tables so I don't think we want to actually mark it as such.